Add new checkmodes

k3nd0x opened the pull request:
- Wireguard
- Services
- Interfaces

Changed plugin output for better viewing in icingaweb2

Commit:
doc: Updated README.md

nbuchwitz commented on check_opnsense.py:
```diff
@@ -27,12 +27,16 @@
 
 import sys
 from typing import Dict, Union
+def jp(data):
```
This method seems to unused -> remove

nbuchwitz commented on check_opnsense.py:
```diff
@@ -70,7 +74,7 @@ def check_output(self) -> None:
         """Print message to stdout and exit with given return code."""
         prefix = rc.name
-        print(f"{prefix} - {message}")
+        print(f"[{prefix}] {message}")
```
Split this change into a separate commit, as this isn't related to the feature

nbuchwitz commented on check_opnsense.py:
```diff
@@ -193,6 +210,10 @@ def parse_args(self) -> None:
             choices=(
                 "updates",
                 "ipsec",
+                "hastatus",
```
for better readability I would call this something like `ha-status` or just `ha`

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,144 @@ def check_ipsec(self) -> None:
         self.perfdata.append(f"tunnels_connected={len(tunnels_connected)}")
         self.perfdata.append(f"tunnels_disconnected={len(tunnels_disconnected)}")
 
+    def check_hastatus(self) -> None:
```
As long this is WIP / not implemented, please move this feature into a separate (draft) PR

nbuchwitz commented on check_opnsense.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Move this change into separate commit (eg "Change python interpreter to python3 explicitly")

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+                interfaces_filtered.append(device)
+        if interfaces_down:
+            counter = len(interfaces_down)
+            self.check_message = f"{counter} Interfaces are down\n"
```
```suggestion
            self.check_message = f"{counter} interface(s) are down\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+            self.check_message += "\n".join(interfaces_down)
+        elif interfaces_up:
+            counter = len(interfaces_up)
+            self.check_message = f"{counter} Interfaces are up\n"
```
```suggestion
            self.check_message = f"{counter} interface(s) are up\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+            self.check_message = f"{counter} Interfaces are up\n"
+
+        for i in interfaces_down:
+            self.check_message += f"[DOWN] Interface {i} is down\n"
```
```suggestion
            self.check_message += f"[DOWN] interface {i} is down\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+            self.check_message += f"[DOWN] Interface {i} is down\n"
+
+        for i in interfaces_up:
+            self.check_message += f"[UP] Interface {i} is up\n"
```
```suggestion
            self.check_message += f"[UP] interface {i} is up\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+        if self.options.verbose >= 1:
+            self.check_message += "\n--- VERBOSE ---\n"
+            for i in interfaces_filtered:
+                self.check_message += f"[FILTER] Interface {i} is filtered by --filter\n"
```
```suggestion
                self.check_message += f"[FILTER] interface {i} is filtered by --filter\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+
+        if failed_services:
+            counter = len(failed_services)
+            self.check_message = f"{counter} Services are failed\n"
```
```suggestion
            self.check_message = f"{counter} services have failed\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+            self.check_result = CheckState.CRITICAL
+        elif running_services:
+            counter = len(running_services)
+            self.check_message = f"{counter} Services are running\n"
```
```suggestion
            self.check_message = f"{counter} services are running\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+            self.check_result = CheckState.OK
+
+        for i in failed_services:
+            self.check_message += f"[CRITICAL] Service {i} is failed\n"
```
```suggestion
            self.check_message += f"[CRITICAL] Service {i} has failed\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+        counter_sum = counter_off + counter_on
+
+        if offline:
+            self.check_message = f"{counter_off}/{counter_sum} Wireguard peers are offline\n"
```
```suggestion
            self.check_message = f"{counter_off}/{counter_sum} WireGuard peers are offline\n"
```

nbuchwitz commented on check_opnsense.py:
```diff
@@ -273,6 +310,146 @@ def check_ipsec(self) -> None:
+            self.check_message = f"{counter_off}/{counter_sum} Wireguard peers are offline\n"
+            self.check_result = CheckState.CRITICAL
+        elif online:
+            self.check_message = f"{counter_on}/{counter_sum} Wireguard peers are online\n"
```
```suggestion
            self.check_message = f"{counter_on}/{counter_sum} WireGuard peers are online\n"
```

nbuchwitz reviewed (changes requested):
Thanks for your contribution of multiple new check features! Please have a look at my review comments and also make sure to run the linter `black` (or `ruff format`) as highlighted in https://github.com/nbuchwitz/check_opnsense/blob/master/CONTRIBUTING.md

If you have any questions how to fix things or the tooling please let me know!

k3nd0x committed:
feat: Change python interpreter to python3

k3nd0x committed:
feat: Make output compatible with IcingaWeb2

k3nd0x committed:
feat: Add wireguard, interface & service checks

k3nd0x commented:
Hi @nbuchwitz,
thanks for your review! I believe I’ve addressed all the points you mentioned.
Let me know if there’s anything else you'd like me to adjust.

nbuchwitz commented:
Sorry for the late response.  I will hopefully find time in the next couple of days to merge this

Commit:
chore: Release version 0.4.0

Signed-off-by: Nicolai Buchwitz <nb@tipi-nte.de>

nbuchwitz commented:
I've squashed the relevant commits and created the 0.4.0 release with your changes. Thanks for your contribution!
